mlir/lib/Dialect/Vector/Transforms/VectorDistribute.cpp

```diff
@@ -526,7 +535,25 @@ struct WarpOpTransferWrite : public OpRewritePattern<WarpExecuteOnLane0Op> {
     // 4. Reindex the write using the distribution map.
     auto newWarpOp =
         newWriteOp.getVector().getDefiningOp<WarpExecuteOnLane0Op>();
+
```

Comment:
It would be nice to have a comment here to say what we are doing in a high level for the following code. Something like if the distributed vector is more than 1-D, we need to delinearize  the 1-D lane ID into the matching dimension.